eknrenderer/eknrenderer/eknr-renderer.c

```diff
@@ -0,0 +1,1377 @@
+
+  return g_variant_new_boolean (FALSE);
+}
+
+static char *
+regex_substitute (const char   *regex,
+                  const char   *substitution,
+                  const char   *content,
+                  GError      **error)
+{
+  g_autoptr(GRegex) regex_compiled = g_regex_new (regex, 0, 0, error);
+
+  if (regex == NULL)
+    return NULL;
+
+  return g_regex_replace (regex_compiled,
+                          content,
+                          -1,
+                          0,
+                          substitution,
+                          0,
+                          error);
+}
+
+static char *
+strip_body_tags (const char  *html,
+                 GError     **error)
+{
+  g_autofree char *stripped_start_tags = regex_substitute ("^\\s*<html>\\s*<body>",
+                                                            "",
+                                                            html,
+                                                            error);
+
+  if (stripped_start_tags == NULL)
+    return NULL;
+
+  return regex_substitute ("<\\/body>\\s*<\\/html>\\s*$",
+                           "",
+                           stripped_start_tags,
+                           error);
+}
+
+static GFile *
+template_file (const char *filename)
+{
+  g_autofree char *uri = g_strdup_printf ("resource:///com/endlessm/knowledge/data/templates/%s", filename);
+  return g_file_new_for_uri (uri);
+}
+
+char *
+_renderer_render_legacy_content (EknrRenderer *renderer,
+                                 const char   *body_html,
+                                 const char   *source,
+                                 const char   *source_name,
+                                 const char   *original_uri,
+                                 const char   *license,
+                                 const char   *title,
+                                 gboolean      show_title,
+                                 gboolean      use_scroll_manager,
+                                 GError      **error)
+{
+  g_autoptr(GFile) file = template_file("legacy-article.mst");
+  g_autofree char *stripped_body = strip_body_tags (body_html, error);
+  GVariantDict vardict;
+  g_autoptr(GVariant) variant = NULL;
+  GVariant *disclaimer = NULL; /* floating */
+
+  if (stripped_body == NULL)
+    return NULL;
+
+  disclaimer = get_legacy_disclaimer_section_content (source,
+                                                      source_name,
+                                                      original_uri,
+                                                      license,
+                                                      title,
+                                                      error);
+
+  if (disclaimer == NULL)
+    return NULL;
+
+  g_variant_dict_init (&vardict, NULL);
+
+  g_variant_dict_insert_value (&vardict,
+                               "title",
+                               show_title ? g_variant_new_string (title) : g_variant_new_boolean (FALSE));
```

Comment:
You probably want an empty string in the false case, no?

Reply:
Not really - the old code inserted a boolean with `false`. This actually translates into "skip this substitution entirely" in mustache-land.